### PR TITLE
Added Support for Non-Standard SSH Ports

### PR DESCRIPTION
**Pull request by paskalito**
Usage: ./hrsync /home/user/Documents /root/Documents root@example.com 666

### CODE REVIEW

**Comment by dparoli** [comment]
Thanks for contributing @paskalito. I think that Instead of `Additions ..` would be better to have `Contributors:` and a list of all contributors, one line each, by username and full name (full name optional). I appreciate also a change in the readme file, under the usage section, to explain how to use the non standard port. If you like it would be better if you changed the PR accordingly, before merging, otherwise I'll fix it later.

**Review by martinhanes — CHANGES_REQUESTED** [review]

File: hrsync [hunk 1]
```diff
@@ -14,6 +15,21 @@ Target="${2%/}"
 TargetHost=$3
 
 # ------------------------------------------------------------
+
+
+# if remote host is set, set SSHPort to user specified value else set it to 22
+if [ -n "$3" ]
+then
+	if [ -z "$4" ]
```
[the remark this answers is not in the document]
Comment (dparoli) [reply hrsync:23]: Well @martinhanes, I agree with the objections you raised and with your proposal. I'll just merge code that implements your solution for non standard ssh ports.

**Comment by paskalito** [comment]
@dparoli you are welcome, and thanks for answering.
I changed the appearance of the contributors as i understood you had proposed it.
feel free to change it if i misunderstood you in any way.
I would also writeup the readme but i think il rather wait with that after it is merged.

File: hrsync (hunk 1 above)
[the remark this answers is not in the document]
Comment (paskalito) [reply hrsync:25]: i had similair thoughts at the start:
(for example the standard rsync syntax for ssh servers is the other way around (which confused me at first) ( root@example.com:/root/Documents))
(this might also be a point why **not** to use the proposed
`./hrsync /home/user/Documents /root/Documents root@example.com:666`
in that way.

However i thought then we should also give proper option flags at least for the ssh option and maybe even input output.
Which would brake existing scripts.

So i thought i just go with the "spirit of the script" and add a 4th parameter.

With that said:
- i also wouldn't know where to start / how to ĉode it with either of the other proposals.
- i do not have any strong opinions or feelings (yet ^^) and personally i also could perfectly live with either of the other Soloutions also if they have an "unconsistent" syntax.

[the remark this answers is not in the document]
Comment (martinhanes) [reply hrsync:25]: Sure, I can see why you would have api which is similar with rsync's api. But if I had to choose betweeen port number as 4th parameter OR the ":" version, both not consistent with rsync, I would go with ":" version every time. That was a motivation behind my comment.

Consistent with rsync or not, hrsync should provide help / comment saying how to specify port number and it's up to the user to provide a port parameter (nor not) in the way hrsync accepts it. **Option flags are in my opinion the best solution** and I agree it would be very nice to be consistent with rsync, however I thought it's not worth the work yet, if we're not planning to add anything else there in foreseeable time :-) 

tl:dr; Providing port after ':' is used in many other tools, and it does not break existing api, and I would go with it.

**Comment by dparoli** [comment]
Sorry for the delay, I had other problems to follow. As I wrote above I would prefer to implement the functionality proposed by @paskalito  in a cleaner way, as proposed by @martinhanes . Right now I don't have the resources to write and / or test a non-standard ssh port solution. Thanks anyway for the proposal.
